builder/dockerfile/builder.go

```diff
@@ -218,7 +225,7 @@ func (b *Builder) build(stdout io.Writer, stderr io.Writer, out io.Writer) (stri
 		for k, v := range b.options.Labels {
 			line += fmt.Sprintf("%q=%q ", k, v)
 		}
-		_, node, err := parser.ParseLine(line)
+		_, node, err := parser.ParseLine(line, &b.directive)
```

Comment:
Not sure about `faster`, but the problem is the node structure isn't passed around in all functions, and more importantly, the directive context is global, not per-node. Trying to do minimal changes, not redesign chunks of the builder...